racket/collects/racket/contract/private/prop.rkt

```diff
@@ -118,7 +118,10 @@
 (define trail (make-parameter #f))
 (define (contract-struct-stronger? a b)
   (cond
-    [(equal? a b) #t]
+    [(and (equal? a b)
+          (or (flat-contract-struct? a)
```

Comment:
Shouldn't this come before `equal?` ?

Reply:
Oh, yeah, that's probably a good change (you're suggesting swapping the order of the arguments to `and`, right?).

Reply:
yes